feynman_path/diagram.py

```diff
@@ -258,3 +289,19 @@ def perform_x(self, q_i, *, pre_latex=f'', name='X'):
             new_state[new_key] += amp
         self.transition_text(self.d, t, f'{pre_latex}{name}_{{{q_i}}}')
         self.add_states(new_state)
+
+    def perform_y(self, q_i, *, pre_latex=f'', name='Y'):
+        new_state = {}
+        t = len(self.state_sequence)-1
+        for key, amp in self.state_sequence[-1].items():
+            is_one = key[q_i] == '1'
+            digits = list(key)
+            digits[q_i] = '01'[not is_one]
+            new_key = ''.join(digits)
+            new_amp = -sympy.I*amp if is_one else sympy.I*amp
+            self.gate_arrow(self.d, t, key, new_key, amp=new_amp/amp)
+            if new_key not in new_state:
+                new_state[new_key] = 0
+            new_state[new_key] += new_amp
+        self.transition_text(self.d, t, f'{pre_latex}{name}_{{{q_i}}}')
+        self.add_states(new_state)
```

Comment:
Nice.  If you intend to add many more gates, can you refactor the perform_* methods to avoid the code copy?

Reply:
It could be as simple as a dictionary with symbolic gate unitaries (at least for single-qubit gates: `{'X': [[0,1],[1,0]], ...}`).  A perform-gate class may just add more boilerplate code.